### PR TITLE
Evaluate Sarah's Tech Challenge Solution

### PR DESCRIPTION
**Pull request by briznad**
Hi @smzajac

I reviewed the functionality of your code challenge submission and you did a stellar job. I appreciate that you went beyond the base requirements and really considered what type of interface would deliver the best user experience by incorporating article images, excerpts, and links.

Justin (@jaymjax1) and I will now go through your code and make comments or ask questions where it seems appropriate. Please don't mind my nitpicking; even though this was a challenge with limited time I'll approach this as I would any code review of a production feature. No expectation to respond, but if you feel inclined to engage, please do.

### CODE REVIEW

**Comment by briznad** [comment]
Oh, by the way, I added your challenge code to a branch on my fork of the code-challenge repo in order to make a pull request, just so we could have this convenient forum to make code comments.

File: new-york-times/src/App.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from "react";
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Fuse from "fuse.js";
```
Comment (jaymjax1) [inline new-york-times/src/App.js:3]: Good choice using fuse.js!

File: new-york-times/src/App.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      try{
+        setLoading(true)
+        const news = await getContent()
+        setNews(news)
+        setLoading(false)
+      }catch(error){
+        setLoading(false)
+        console.error(error)
+      }
+    })();
```
Comment (jaymjax1) [inline new-york-times/src/App.js:28]: Nice clever usage of an IIFE inside the useEffect hook!

File: new-york-times/src/App.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+     { !loading ?
+      <>
+        <header>
+          <Search onChange={(e) => (setFilter(e.target.value))}/>
+        </header>
+        <Articles news={filtered}/>
+      </>
+      :
+      <CircularProgress size={200} thickness={2} style={{marginTop: "20%"}}/>
+    }
```
Comment (jaymjax1) [inline new-york-times/src/App.js:65]: Glad to see a loading state here, with a reusable component as well

File: new-york-times/src/Components/Articles.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      width: 345,
+      height: "100%",
+      display: "flex",
+      flexDirection: "column",
+      justifyContent: "space-between"
+    },
+    media: {
+      height: 140,
+    },
+  });
```
Comment (jaymjax1) [inline new-york-times/src/Components/Articles.js:22]: The `makeStyles` hook is pretty neat. Curious of you opinions of an external stylesheet vs this implementation?

File: new-york-times/src/Components/Search.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            onChange={props.onChange}
+            InputProps={{
+            startAdornment: (
+                <InputAdornment position="start">
+                <img alt="search" style={{height: "50px"}} src={searchIcon}/>
+                </InputAdornment>
+            ),
+            }}  
+            />
+        </>
```
Comment (jaymjax1) [inline new-york-times/src/Components/Search.js:23]: So when do you decide to use React Fragments vs not using them?

File: new-york-times/src/content.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios';
+
+async function getContent(){
+      let res =  await axios.get('https://api.nytimes.com/svc/topstories/v2/science.json?api-key=Gwxln5M3geWlhR6UE0TY1FUWKSG3wCil')
+
+        //fun extra asyncness to showcase loading phase
+        await new Promise(res => setTimeout(res, 1000))
```
Comment (jaymjax1) [inline new-york-times/src/content.js:7]: Thanks for showcasing this!

**Pull request closed without merging**